transport/http/handler.go

```diff
@@ -59,6 +59,11 @@ func (h handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if err == context.DeadlineExceeded {
+		http.Error(w, "remote timeout", http.StatusGatewayTimeout)
```

Comment:
The advantage of doing it this way is that you get better, readable error messages even when the client is not a YARPC client.